src/main/java/com/minecolonies/tileentities/TileEntityColonyBuilding.java

```diff
@@ -229,13 +227,12 @@ public void setBuilding(AbstractBuilding b)
     public NBTTagCompound writeToNBT(@NotNull NBTTagCompound compound)
     {
         super.writeToNBT(compound);
+        /*
         if (colonyId == 0 && colony == null)
         {
-            colony = ColonyManager.getColony(worldObj, this.getPosition());
-            Log.getLogger().fatal(String.format("TileEntityColonyBuilding at %s:[%d,%d,%d] has no colonyId; %s colony reference.",
-              worldObj.getWorldInfo().getWorldName(), pos.getX(), pos.getY(), pos.getZ(),
-              colony == null ? "NO" : "valid"));
+            //todo: actually do something about it and not spam the server
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1135)